awx/main/middleware.py

```diff
@@ -26,6 +26,11 @@
 perf_logger = logging.getLogger('awx.analytics.performance')
 
 
+class SettingsCacheMiddleware(MiddlewareMixin):
+    def process_request(self, request):
+        settings._awx_conf_memoizedcache.clear()
```

Comment:
I don't get the reasoning of doing this here.

Reply:
Any given automated script could do this:
 - make a PATCH to `/api/v2/settings/jobs/`, applying new values
 - right after that, make a GET to `/api/v2/settings/jobs/`

Regardless of what uWSGI worker processes the second (GET) request, it will be under the 5 second `ttl` threshold. That means that it would give the values from before the PATCH. The automated script will then conclude that the PATCH it made did not take effect.

Now replace the generalized "automated script" with out integration tests, and you see the immediate reason this was needed.